### DISCUSSION TITLE
Why does DefaultRequest append a space after the endpoint?

### DISCUSSION THREAD

**Discussion by SmarMykel** [aws/aws-sdk-java#2960] [General]
```
@Override
    public String toString() {
        final StringBuilder builder = new StringBuilder();
        builder.append(getHttpMethod()).append(" ");
        builder.append(getEndpoint()).append(" ");
        String resourcePath = getResourcePath();

        if (resourcePath == null) {
            builder.append("/");
        }
        else {
            if (!resourcePath.startsWith("/")) {
                builder.append("/");
            }
            builder.append(resourcePath);
        }

        return builder.toString();
    }
```

    
 In [aws-sdk-java/aws-java-sdk-core/src/main/java/com/amazonaws/DefaultRequest.java](https://github.com/aws/aws-sdk-java/blob/master/aws-java-sdk-core/src/main/java/com/amazonaws/DefaultRequest.java) toString method does this. I have been trying to understand why this necessary but can't understand.
    
Why is there a space after the  `getEndpoint()` ?. 
Specifically  `builder.append(" ")`  after `builder.append(getEndpoint())`
    
 If my endpoint is `Endpoint: https://api.example.com` and `Resource Path: /users/123`, then I will get `GET https://api.example.com /users/123`
    
 Am I missing something?

**Comment by debora-ito** [comment]
Duplicate of https://github.com/aws/aws-sdk-java/issues/2962.
Closing.
